lgc/state/PipelineState.cpp

```diff
@@ -553,12 +553,21 @@ const ResourceNode *PipelineState::findPushConstantResourceNode() const {
 
 // =====================================================================================================================
 // Find the resource node for the given {set,binding}.
-// For nodeType == Unknown, the function finds any node of the given set,binding.
-// For nodeType == Resource, it matches Resource or CombinedTexture.
-// For nodeType == Sampler, it matches Sampler or CombinedTexture.
-// For nodeType == Buffer, it matches Buffer, BufferCompact or InlineBuffer.
-// For other nodeType, only a node of the specified type is returned.
-// For nodeType == DescriptorTableVaPtr, the node is returned whose first child matches descSet.
+// A node type (A) is compatible with a node type (B) iff (A) <= (B) in the ResourceNodeType lattice:
```

Comment:
Also, I think this definition of "compatible" is too lax. Asking for DescriptorYCbCrSampler should only give you DescriptorYCbCrSampler. Asking for DescriptorTexelBuffer should only give you DescriptorTexelBuffer. However, it is difficult to imagine a case where that would actually make it go wrong on real world code.

Reply:
This PR was supposed to be a non-functional change, so I preserved the original definition.
If it makes sense, I'll change the logic like you suggest.

Reply:
Seems like YCbCrSampler is not more now anyway.